### PR TITLE
Postgres alert improvements

### LINKED ISSUE #787: Alerts on lagging/unknown postgres member

### LINKED ISSUE #802: upgrade postgresql exporter

### LINKED ISSUE #811: pg logs to stderr
Postgres logs should not redirect to files on disk and should log straight to stderr

### PR DESCRIPTION
**Pull request by BrendanGalloway**
### Description

Update platform operator version to 1.6.2 and reverted to upstream image to enable exporter configmap volume mount
Add replication slot info metric to exporter configmap
Upgrade postgres exporter version to 0.9.0
Added annotations and config to export postgres logs to elastic

### Dependencies
NA

### Breaking Change

- [ ] Yes
- [x] No

### Testing Notes

Set up kind cluster  with postgres operator
Created zalando object:
```
apiVersion: acid.zalan.do/v1
kind: postgresql
metadata:
  name: postgres-test
  namespace: postgres-operator
spec:
  additionalVolumes:
  - mountPath: /opt/extra-queries
    name: postgres-exporter-config
    subPath: ""
    targetContainers:
    - exporter
    volumeSource:
      configMap:
        name: postgres-exporter-config
  allowedSourceRanges: null
  dockerImage: docker.io/flanksource/spilo:1.6-p2.flanksource
  enableShmVolume: true
  env:
  - name: USE_WALG_RESTORE
    value: "false"
  - name: USE_WALG_BACKUP
    value: "false"
  numberOfInstances: 3
  patroni:
    initdb:
      data-checksums: "true"
      encoding: UTF8
      locale: en_US.UTF-8
    loop_wait: 10
    maximum_lag_on_failover: 33554432
    pg_hba:
    - hostssl all all 0.0.0.0/0 md5
    - host    all all 0.0.0.0/0 md5
    retry_timeout: 10
    slots: {}
    synchronous_mode: false
    synchronous_mode_strict: false
    ttl: 30
  podAnnotations: {}
  postgresql:
    parameters:
      archive_mode: "on"
      archive_timeout: 60s
      log_destination: stderr
      max_connections: "600"
      shared_buffers: 2048MB
    version: "12"
  resources:
    limits:
      cpu: 1000m
      memory: 512Mi
    requests:
      cpu: 100m
      memory: 128Mi
  serviceAnnotations: {}
  sidecars:
  - env:
    - name: PG_EXPORTER_WEB_LISTEN_ADDRESS
      value: :9187
    - name: DATA_SOURCE_URI
      value: localhost?sslmode=disable
    - name: DATA_SOURCE_USER
      valueFrom:
        secretKeyRef:
          key: username
          name: postgres.postgres-test.credentials
    - name: DATA_SOURCE_PASS
      valueFrom:
        secretKeyRef:
          key: password
          name: postgres.postgres-test.credentials
    - name: PG_EXPORTER_EXTEND_QUERY_PATH
      value: /opt/extra-queries/queries.yaml
    image: docker.io/bitnami/postgres-exporter:0.9.0
    name: exporter
    ports:
    - containerPort: 9187
      name: exporter
      protocol: TCP
    resources:
      limits:
        cpu: 200m
        memory: 128Mi
      requests:
        cpu: 10m
        memory: 128Mi
  teamId: postgres
  tls: null
  users:
    app:
    - createdb
    - superuser
  volume:
    size: 5Gi
```
verified configmap volumemount is created
set up port-forwarding to metrics server:
```
kubectl port-forward postgres-test-0 9187:9187
```
verified new metrics are reported:
```
# HELP pg_replication_slot_status_out Dummy value for metric to filter on
# TYPE pg_replication_slot_status_out gauge
pg_replication_slot_status_out{active="true",server="localhost:5432",slot_name="postgres_test_1"} 1
pg_replication_slot_status_out{active="true",server="localhost:5432",slot_name="postgres_test_2"} 1
```

Created postgresqldb object and repeated the above steps:

```
apiVersion: db.flanksource.com/v1
kind: PostgresqlDB
metadata:
  name: test
  namespace: postgres-operator
spec:
  resources:
    limits:
      cpu: "2"
      memory: 2Gi
  parameters:
    max_connections: "600"
    shared_buffers: 4096MB
  replicas: 2
  storage:
    size: 5Gi
```
Verified that new template outputs logs to stderr:

 ```
 kubectl logs postgres-test-1 postgres -n postgres-operator --tail 10 -f
 2021-04-19 13:11:19.352 UTC [103867] [unknown]@[unknown] LOG:  connection received: host=[local]
2021-04-19 13:11:19.352 UTC [103867] postgres@test LOG:  connection authorized: user=postgres database=test application_name=psql
2021-04-19 13:11:19.356 UTC [103867] postgres@test LOG:  disconnection: session time: 0:00:00.004 user=postgres database=test host=[local]
2021-04-19 13:11:19.389 UTC [103876] [unknown]@[unknown] LOG:  connection received: host=[local]
2021-04-19 13:11:19.389 UTC [103876] postgres@test LOG:  connection authorized: user=postgres database=test application_name=psql
2021-04-19 13:11:19.392 UTC [103876] postgres@test LOG:  disconnection: session time: 0:00:00.003 user=postgres database=test host=[local]
2021-04-19 13:11:19.427 UTC [103885] [unknown]@[unknown] LOG:  connection received: host=[local]
2021-04-19 13:11:19.427 UTC [103885] postgres@test LOG:  connection authorized: user=postgres database=test application_name=psql
2021-04-19 13:11:19.430 UTC [103885] postgres@test LOG:  disconnection: session time: 0:00:00.003 user=postgres database=test host=[local]
```

### **Links**

Fixes #787
Fixes #802
Fixes #811 
Causes #828

### CODE REVIEW

File: manifests/postgres-operator.yaml [hunk 1]
```diff
@@ -269,7 +269,7 @@ spec:
       serviceAccountName: postgres-operator
       containers:
         - name: postgres-operator
-          image: docker.io/flanksource/postgres-operator:{{.postgresOperator.version}}
+          image: registry.opensource.zalan.do/acid/postgres-operator:{{.postgresOperator.version}}
```
Comment (moshloop) [inline manifests/postgres-operator.yaml:272]: I think we need to mirror this to flanksouree dockerhub as I don't think registry.opensource.zalan.do is a production-grade registry

**Review by moshloop — CHANGES_REQUESTED** [review]

File: manifests/postgres-operator.yaml (hunk 1 above)
Comment (BrendanGalloway) [reply manifests/postgres-operator.yaml:272]: Can do - would you like me to mirror a range of releases, or just 1.6.2 ?

File: manifests/postgres-operator.yaml [hunk 2]
```diff
@@ -269,7 +269,7 @@ spec:
       serviceAccountName: postgres-operator
       containers:
         - name: postgres-operator
-          image: docker.io/flanksource/postgres-operator:{{.postgresOperator.version}}
+          image: flanksource/postgres-operator:{{.postgresOperator.version}}
```
Comment (moshloop) [inline manifests/postgres-operator.yaml:272]:
```suggestion
          image: docker.io/flanksource/postgres-operator:{{.postgresOperator.version | default "na" }}
```

File: test/postgres.yaml [hunk 3]
```diff
@@ -17,10 +17,18 @@ s3:
...
   defaultBackupRetention:
     keepLast: 5
     keepHourly: 2
-    keepDaily: 1
\ No newline at end of file
+    keepDaily: 1
+thanos:
+  version: v0.13.0
+  mode: client
+  bucket: thanos
```
Comment (moshloop) [inline test/postgres.yaml:30]:
```suggestion
```

File: manifests/postgres-operator.yaml (hunk 2 above)
Comment (BrendanGalloway) [reply manifests/postgres-operator.yaml]: karina defaults in an operator version if not supplied (I bumped to 1.6.2 in this change set) - would it not be better to have the same value here?

File: pkg/phases/postgresoperator/install.go [hunk 4]
```diff
@@ -40,5 +40,5 @@ func Deploy(platform *platform.Platform) error {
 		return err
 	}
 
-	return platform.ApplySpecs(Namespace, "postgres-operator.yaml", "postgres-exporter-config.yaml", "postgres-operator-config.yaml", "template/postgresql-db.yaml.raw")
+	return platform.ApplySpecs(Namespace, "postgres-operator.yaml", "postgres-operator-monitoring.yaml", "postgres-exporter-config.yaml", "postgres-operator-config.yaml", "template/postgresql-db.yaml.raw")
```
Comment (moshloop) [inline pkg/phases/postgresoperator/install.go:43]:
```suggestion
	return platform.ApplySpecs(Namespace, "postgres-operator.yaml", "postgres-operator-monitoring.yaml.raw", "postgres-exporter-config.yaml", "postgres-operator-config.yaml", "template/postgresql-db.yaml.raw")
```

**Pull request closed without merging**